app/code/Magento/Store/Model/StoreRepository.php

```diff
@@ -104,13 +106,15 @@ public function getById($id)
         }
 
         $storeData = $this->getAppConfig()->get('scopes', "stores/$id", []);
-        $store = $this->storeFactory->create([
-            'data' => $storeData
-        ]);
+        $store = $this->storeFactory->create(
+            [
+                'data' => $storeData
+            ]
+        );
 
         if ($store->getId() === null) {
             throw new NoSuchEntityException(
-                __("The store that was requested wasn't found. Verify the store and try again.")
+                __("The store with id " . (int)$id . " that was requested wasn't found. Verify the store and try again.")
```

Comment:
This will cause translations to fail unless the number is hardcoded in the translations, it should probably be `__("The store with id that was requested [%u] wasn't found. Verify the store and try again.", (int)$id)`

Reply:
Good catch, thank you!